letta/agent.py

```diff
@@ -1333,6 +1334,50 @@ def retry_message(self) -> List[Message]:
         assert all(isinstance(msg, Message) for msg in messages), "step() returned non-Message objects"
         return messages
 
+    def get_context_window(self) -> ContextWindowOverview:
+        """Get the context window of the agent"""
+
+        system_prompt = self.agent_state.system  # TODO is this the current system or the initial system?
```

Comment:
@sarahwooders I recall something like:
`agent_state.system` is the initial state of the system prompt
and `agent_state.state["system"]` is the latest state of the system prompt

but this didn't seem to match up with the current code? maybe something changed and I forgot (is the latest system prompt stored inside `Memory`)?

Reply:
Yeah I've been using `agent_state.system`, since I try to avoid dictionaries